pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -528,6 +544,7 @@ func (a *ipAssigner) addVLANAssignee(link netlink.Link, vlan int32) (*assignee,
 		logicalInterface: iface,
 		link:             link,
 		ips:              sets.New[string](),
+		advertiseFn:      advertiseFnc,
```

Comment:
To replace advertise from test, specially for non vlan case

Reply:
It doesn't explain why it's set to a dummy function here. This is not test code.

Reply:
To mock advertise( replace defer and restore strategy for other apis was followed as mentioned in netlink, but not for advertise), advertise function is added in assignee struct, line 73 , advertiseFn func(ip net.IP, externalInterface *net.Interface), since vlan and non vlan use the structure assigne, a dummy function shld be added to avoid invalid memory address or nil pointer dereference

Reply:
This is wrong. both vlan and non-vlan interfaces need to advertise.

Reply:
advertise api syntax changed, now it is rebased